framework/src/service/ServiceReferenceBasePrivate.h

```diff
@@ -124,6 +130,153 @@ namespace cppmicroservices
         InterfaceMapConstPtr GetServiceFromFactory(BundlePrivate* bundle,
                                                    std::shared_ptr<ServiceFactory> const& factory);
     };
+
+    /* @brief Private helper struct used to facilitate the shared_ptr aliasing constructor
+     *        in BundleContext::GetService method. The aliasing constructor helps automate
+     *        the call to UngetService method.
+     *
+     *        Service consumers can simply call GetService to obtain a shared_ptr to the
+     *        service object and not worry about calling UngetService when they are done.
+     *        The UngetService is called when all instances of the returned shared_ptr object
+     *        go out of scope.
+     */
+    template <class S>
+    struct ServiceHolder
```

Comment:
`ServiceHolder` and `UngetHelper` enable the same behavior for two different use cases (singleton service vs. bundle/prototype service). Now that these classes are in the same file, does it make sense to combine these two classes into one class that handles both use cases?

Combining these classes isn't required and I am fine with doing this at a later time if its possible and desirable.